packages/backend/src/managers/llama-stack/llamaStackManager.ts

```diff
@@ -0,0 +1,251 @@
+
+    this.createLlamaStackContainer(connection, labels)
+      .then((containerInfo: LlamaStackContainerInfo) => {
+        this.#containerInfo = containerInfo;
+        this.taskRegistry.updateTask({
+          ...task,
+          state: 'success',
+          labels: {
+            ...task.labels,
+            containerId: containerInfo.containerId,
+            port: `${containerInfo.port}`,
+          },
+        });
+        this.telemetryLogger.logUsage('llamaStack.startContainer');
+      })
+      .catch((err: unknown) => {
+        // Get all tasks using the tracker
+        const tasks = this.taskRegistry.getTasksByLabels({
+          trackingId: trackingId,
+        });
+        // Filter the one no in loading state
+        tasks
+          .filter(t => t.state === 'loading' && t.id !== task.id)
+          .forEach(t => {
+            this.taskRegistry.updateTask({
+              ...t,
+              state: 'error',
+            });
+          });
+        // Update the main task
+        this.taskRegistry.updateTask({
+          ...task,
+          state: 'error',
+          error: `Something went wrong while trying to create an inference server ${String(err)}.`,
+        });
+        this.telemetryLogger.logError('llamaStack.startContainer', { error: err });
+      });
+  }
+
+  async createLlamaStackContainer(
+    connection: ContainerProviderConnection,
+    labels: { [p: string]: string },
+  ): Promise<LlamaStackContainerInfo> {
+    const image = llama_stack_images.default;
+    const pullingTask = this.taskRegistry.createTask(`Pulling ${image}.`, 'loading', labels);
+    const imageInfo = await getImageInfo(connection, image, () => {})
+      .catch((err: unknown) => {
+        pullingTask.state = 'error';
+        pullingTask.progress = undefined;
+        pullingTask.error = `Something went wrong while pulling ${image}: ${String(err)}`;
+        throw err;
+      })
+      .then(imageInfo => {
+        pullingTask.state = 'success';
+        pullingTask.progress = undefined;
+        return imageInfo;
+      })
+      .finally(() => {
+        this.taskRegistry.updateTask(pullingTask);
+      });
+
+    const folder = await this.getLlamaStackContainerFolder();
+
+    const containerTask = this.taskRegistry.createTask('Starting Llama Stack container', 'loading', labels);
+    const aiLabApiPort = this.configurationRegistry.getExtensionConfiguration().apiPort;
+    const llamaStackApiPort = await getFreeRandomPort('0.0.0.0');
+    const createContainerOptions: ContainerCreateOptions = {
+      Image: imageInfo.Id,
+      name: getRandomName('llama-stack'),
+      Labels: {
+        [LLAMA_STACK_CONTAINER_LABEL]: image,
+        [LLAMA_STACK_API_PORT_LABEL]: `${llamaStackApiPort}`,
+      },
+      HostConfig: {
+        AutoRemove: true,
+        SecurityOpt: [DISABLE_SELINUX_LABEL_SECURITY_OPTION],
+        Mounts: [
+          {
+            Target: '/app/.llama',
+            Source: path.join(folder, '.llama'),
```

Comment:
You seems to only use `getLlamaStackContainerFolder` once (L191) and you join the path to `.llama`

I don't understand why not just returning the `.llama`? why do we need the parent folder?